### PR TITLE
Sarah afiya/issue 17/fix issues in matching code examples with doc api

### PR DESCRIPTION
**Pull request by sarah-afiya**
Wrote test cases for all the failing cases and refactored code to satisfy those

### CODE REVIEW

File: documentation_quality_analysis/analyze_library/signature_matcher/python_signature_matcher.py [hunk 1]
```diff
@@ -42,31 +42,29 @@ def python_match_examples(repo_name: str,
 
                 if len(function_split) > 1:
                     if function_split[0] in var_declarations:
```
Comment (snadi) [inline documentation_quality_analysis/analyze_library/signature_matcher/python_signature_matcher.py:44]: I'd suggest providing a meaningful name to `function_split[0]` ?

**Review by snadi — APPROVED** [review]
Nice work adding those tests!

File: documentation_quality_analysis/analyze_library/signature_matcher/python_signature_matcher.py (hunk 1 above)
Comment (sarah-afiya) [reply documentation_quality_analysis/analyze_library/signature_matcher/python_signature_matcher.py:44]: so, a call can be like this -> call = pd.Series.str.add 
and then, function_split = ['pd', 'Series', 'str', 'add']
Should I call it called_names or something like that?

Comment (snadi) [reply documentation_quality_analysis/analyze_library/signature_matcher/python_signature_matcher.py:44]: but isn't `function_split[0]` then intended to be a var_decl of some sort? 
So I would just name `function_split` as `statement_parts` and `function_split[0]` as `var_decl`
if that makes sense

Comment (sarah-afiya) [reply documentation_quality_analysis/analyze_library/signature_matcher/python_signature_matcher.py:44]: It does. Thanks!

**Pull request merged**